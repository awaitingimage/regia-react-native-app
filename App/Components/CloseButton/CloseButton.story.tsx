import { storiesOf } from "@storybook/react-native";
import * as React from "react";

import CloseButton from "./CloseButton";

storiesOf("CloseButton", module)
  .add("Default", () => (
    <CloseButton />
  ));
