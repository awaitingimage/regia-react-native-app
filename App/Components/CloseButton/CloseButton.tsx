import * as React from "react";
import { GestureResponderEvent, StyleProp, TouchableOpacity, View, ViewStyle } from "react-native";
import Icon from "react-native-vector-icons/Ionicons";
import { Colors } from "../../Themes/";
import styles from "./Style";

interface Props {
  style?: StyleProp<ViewStyle>;
  onPress?: (event: GestureResponderEvent) => void;
  color?: string;
}

const defaultProps = {
  style: styles.container,
  onPress: () => alert("back button"),
  color: "black",
};

const CloseButton: React.SFC<Props> = ({style, onPress, color}: Props) =>  (
  <TouchableOpacity accessibilityLabel={"close"} onPress={onPress} style={style}>
    <Icon name="md-close" size={34} color={color} />
  </TouchableOpacity>
);

CloseButton.defaultProps = defaultProps;

export default CloseButton;
