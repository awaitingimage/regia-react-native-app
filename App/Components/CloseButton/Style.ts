import { StyleSheet } from "react-native";
import { ApplicationStyles, Colors, Fonts, Metrics } from "../../Themes/";

export default StyleSheet.create({
  container: {
    alignSelf: "flex-end",
    paddingTop: 20,
    paddingRight: 20,
  },
});
