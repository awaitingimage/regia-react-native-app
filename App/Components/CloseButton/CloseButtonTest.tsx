/// <reference types="@types/jest" />
import * as React from "react";
import * as renderer from "react-test-renderer";
import CloseButton from "./CloseButton";

test("CloseButton component renders correctly", () => {
  const tree = renderer.create(<CloseButton />).toJSON();
  expect(tree).toMatchSnapshot();
});
