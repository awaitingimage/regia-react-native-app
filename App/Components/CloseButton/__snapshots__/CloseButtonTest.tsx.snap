// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CloseButton component renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel="close"
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignSelf": "flex-end",
      "opacity": 1,
      "paddingRight": 20,
      "paddingTop": 20,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "black",
          "fontSize": 34,
        },
        undefined,
        Object {
          "fontFamily": "Ionicons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }
  >
    ï‹€
  </Text>
</View>
`;
